plugins/carbon_black_cloud/icon_carbon_black_cloud/tasks/monitor_alerts/task.py

```diff
@@ -0,0 +1,258 @@
+
+    def get_alerts(
+        self, start_alert_time: str, end_alert_time: str, state: Dict[str, str]
+    ) -> Tuple[list, bool, Dict[str, str]]:
+        alerts_has_more_pages = False
+        endpoint = f"api/alerts/v7/orgs/{self.connection.org_key}/alerts/_search"
+        url = f"{self.connection.base_url}/{endpoint}"
+
+        payload = {
+            "time_range": {"start": start_alert_time, "end": end_alert_time},
+            "criteria": {},
+            "start": "1",
+            "rows": str(PAGE_SIZE),  # max number of results that can be returned
+            "sort": [{"field": ALERT_TIME_FIELD, "order": "ASC"}],
+        }
+        self.logger.info(f"Querying alerts using parameters {payload['time_range']}")
+        resp = self.connection.request_api(url, payload)
+
+        alerts = resp.get("results", [])
+        if alerts:
+            # Check if we have not got all available alerts otherwise trigger task again to catch up quicker
+            # CB can return max 10K during on time frame in which case we need to shorten the frame
+            if resp.get("num_found", 0) > PAGE_SIZE:
+                self.logger.info("Have not got all alerts for given time period. Returning has more pages true...")
+                alerts_has_more_pages = True
+
+            alerts, state = self._dedupe_and_get_last_time(alerts, state, start_alert_time, observations=False)
+        else:
+            self.logger.info("No alerts retrieved for time period searched...")
+            state[LAST_ALERT_TIME] = end_alert_time
+        self.logger.info(f"{LAST_ALERT_TIME} set to: {state[LAST_ALERT_TIME]}, has_more_pages={alerts_has_more_pages}")
+
+        return alerts, alerts_has_more_pages, state
+
+    def trigger_observation_search_job(self, start_time: str, end_time: str, state: Dict[str, str]) -> Tuple[str, Dict]:
+        endpoint = f"api/investigate/v2/orgs/{self.connection.org_key}/observations/search_jobs"
+        search_params = {
+            "rows": PAGE_SIZE,
+            "start": 0,
+            "fields": ["*"],
+            "criteria": {"observation_type": OBSERVATION_TYPES},
+            "sort": [{"field": OBSERVATION_TIME_FIELD, "order": "asc"}],
+            "time_range": {"start": start_time, "end": end_time},
+        }
+        url = f"{self.connection.base_url}/{endpoint}"
+        self.logger.info(f"Triggering observation search using parameters {search_params['time_range']}")
+        observation_job_id = self.connection.request_api(url, search_params).get("job_id")
+
+        if observation_job_id:
+            self.logger.info(
+                f"Saving observation job ID {observation_job_id} to the state. "
+                f"Will query this after polling for alerts..."
+            )
+            state[LAST_OBSERVATION_JOB] = observation_job_id
+
+            if not state.get(LAST_OBSERVATION_TIME):
+                # We should only hit this when we have *never* returned any observations, but after we've
```

Comment:
Maybe reword this comment as its a little confusing?

Reply:
Yeah totally agree, even I'm questioning what I meant by this 😆 will update!